### ISSUE TITLE
iphone scale issue...

### ISSUE THREAD

**Issue by chrisribe** [matiasgagliano/guillotine#61]
It seems that under ios, picking a photo that was taken in portrait mode messes up the scaling on display.
What probably happens is that ios rotates the display image (I think), since once cropped and uploaded to server the picture ends up 90 degrees off and squished...

You can see something is wrong in the crop view (the image is off scale).
Here are the example images.

**Original ios (holding phone in portrait mode)**
![1_goodscale](https://cloud.githubusercontent.com/assets/1999791/21772526/af9a1556-d659-11e6-887f-856ff958a625.jpg)

**Display in guillotine squished under ios (holding phone in portrait mode)**
![2_badscale](https://cloud.githubusercontent.com/assets/1999791/21772525/af6bc2e6-d659-11e6-85ad-c400ddf15e67.jpg)

Any ideas on how to fix / handle this ?
Thanks
Chris

**Comment by chrisribe** [comment]
Iphone EXIF rotation seems to be the issue, ios rotates the image for proper display but saves the rotation in EXIF. The actual image data is not rotated so guillotine crop data does not fit.

- http://stackoverflow.com/questions/1041884/iphone-camera-images-are-rotated-when-uploaded-to-web?noredirect=1&lq=1
- http://stackoverflow.com/questions/18519160/exif-image-rotation-issue-using-carrierwave-and-rmagick-to-upload-to-s3?noredirect=1&lq=1

Looking for solution...

**Comment by chrisribe** [comment]
Found a solution,

 I simply apply the exif Orientation instructions on the server side after upload. This is so that the uploaded photo from the server is oriented properly BEFORE display in guillotine.

Under IOS the phone seems to display exif photos in their proper orientation. This messes up guillotine since the given crop parameters are invalid for the source file.

I applied this function to my uploaded picture and re-save the file.
http://php.net/manual/fr/function.imagecreatefromjpeg.php#112902

Chris
